internal/cmd/run/cmd.go

```diff
@@ -118,7 +121,19 @@ func (r *run) run(cmd *cobra.Command) {
 		LeaderElectionID:           r.leaderElectionID,
 		LeaderElectionNamespace:    r.leaderElectionNamespace,
 		LeaderElectionResourceLock: resourcelock.ConfigMapsResourceLock,
-		ClientBuilder:              manager.NewCachingClientBuilder(),
+		NewClient: func(cache cache.Cache, config *rest.Config, options client.Options, uncachedObjects ...client.Object) (client.Client, error) {
+			// Create the client for Write operation
+			c, err := client.New(config, options)
+			if err != nil {
+				return nil, err
+			}
+			return client.NewDelegatingClient(client.NewDelegatingClientInput{
+				CacheReader:       cache,
+				Client:            c,
+				UncachedObjects:   uncachedObjects,
+				CacheUnstructured: true,
+			})
+		},
```

Comment:
To make this re-usable and to de-clutter this file, what do you think about keeping `pkg/manager/delegatingclient.go` around and defining a function there that returns a function with the `NewClientFunc` signature.

It may even be possible (though maybe not necessary) to keep the existing code around (maybe deprecate it) and have this new function just wrap the existing code.

Something like:
```go
func NewCachingClientFunc() cluster.NewClientFunc {
	return func(cache cache.Cache, config *rest.Config, options client.Options, uncachedObjects ...client.Object) (client.Client, error)
		return NewCachingClientBuilder().WithUncached(uncachedObjects...).Build()
	}
}
```

This approach would let you keep the tests for this around in an easier-to-maintain way as well.

Reply:
@joelanford the previous implementation returned `manager.ClientBuilder` which has been [removed](https://github.com/kubernetes-sigs/controller-runtime/pull/1409) from in controller-runtime, so we can't keep the existing code. I'm not sure of creating a new function which wraps the existing `NewClientFunc` without keeping the previous code (since we have to remove it) as its already defined [here](https://github.com/kubernetes-sigs/controller-runtime/blob/c0a5babc6854d42ab37eb5323ff024eb0dc24b9d/pkg/cluster/cluster.go#L256).

Wouldn't the wrapper still look like what it is now:
```

func NewCachingClient(cache cache.Cache, config *rest.Config, options client.Options, uncachedObjects ...client.Object) (client.Client, error) {
	c, err := client.New(config, options)
	if err != nil {
		return nil, err
	}
	return client.NewDelegatingClient(client.NewDelegatingClientInput{
		CacheReader:       cache,
		Client:            c,
		UncachedObjects:   uncachedObjects,
		CacheUnstructured: true,
	})
}
```